consensus/core/src/test_dag_builder.rs

```diff
@@ -781,6 +801,24 @@ impl<'a> LayerBuilder<'a> {
         }
     }
 
+    fn block_timestamp(
+        &self,
+        authority: AuthorityIndex,
+        round: Round,
+        num_block: u32,
+    ) -> BlockTimestampMs {
+        if self.specified_authorities.is_some() && !self.timestamps.is_empty() {
+            let specified_authorities = self.specified_authorities.as_ref().unwrap();
+
+            if let Some(position) = specified_authorities.iter().position(|&x| x == authority) {
+                return self.timestamps[position] + (round + num_block) as u64;
```

Comment:
Since timestamps are specified for each layer, is it really necessary to add the round and num_block as "jitters"?

Reply:
A layer can still contain multiple rounds, that's why we need to offset them.